Tests/ApolloPerformanceTests/ParsingPerformanceTests.swift

```diff
@@ -0,0 +1,58 @@
+import XCTest
+@testable import Apollo
+import ApolloTestSupport
+import GitHubAPI
+
+class ParsingPerformanceTests: XCTestCase {
+  
+  override func setUpWithError() throws {
+  }
+  
+  override func tearDownWithError() throws {
+  }
+  
+  func testParseResult() throws {
+    let query = IssuesAndCommentsForRepositoryQuery()
+    
+    let response = try loadResponse(for: query)
+    
+    measure {
+      whileRecordingErrors {
+        let (result, _) = try response.parseResult().await()
+        
+        let data = try XCTUnwrap(result.data)
+        XCTAssertEqual(data.repository?.name, "apollo-ios")
+      }
+    }
+  }
+  
+  func testParseResultFast() throws {
+    let query = IssuesAndCommentsForRepositoryQuery()
+    
+    let response = try loadResponse(for: query)
+    
+    measure {
+      whileRecordingErrors {
+        let result = try response.parseResultFast()
+        
+        let data = try XCTUnwrap(result.data)
+        XCTAssertEqual(data.repository?.name, "apollo-ios")
+      }
+    }
+  }
+  
+  // MARK - Helpers
+  
+  func loadResponse<Query: GraphQLQuery>(for query: Query, file: StaticString = #file, line: UInt = #line) throws -> GraphQLResponse<Query.Data> {
+    let bundle = Bundle(for: type(of: self))
+    
+    guard let url = bundle.url(forResource: query.operationName, withExtension: "json") else {
+      throw XCTSkip("Missing response file for query: \(query.operationName)", file: file, line: line)
```

Comment:
I think this should be an `XCTFail` rather than `XCTSkip` - I'd rather have that be real obvious real quick.

Reply:
I agree, but the problem is that `XCTFail` doesn't throw, so we'd need a custom error just to stop execution. Maybe that's worth it? Not sure what to call it. Maybe `throw XCTFailWithError()`?

Reply:
Hmmm, I tried adding a `XCTPreconditionFailure`, but I don't know how to replicate how `XCTSkip` allows you to pass in `file` and `name` to still associate a thrown error with the test that called the helper method.

Reply:
Are you ok leaving `XCTSkip` in here for now?

Reply:
I figured out a solution, so I added `XCTFailure` as an analog of `XCTSkip` that fails the test instead of marking it as skipped. Went with `XCTFailure` instead of something more specific like `XCTPreconditionFailure` because ending execution early seems generally useful, and this is basically an error you can throw to invoke `XCTFail` and end the test.